File: from-2401-to-2500/2490-circular-sentence/solution.go
func isCircularSentence(sentence string) bool {
    if sentence[0] != sentence[len(sentence)-1] {
        return false
    }
    words := strings.Fields(sentence)
    for i := 0; i < len(words) - 1; i++ {
        if words[i][len(words[i])-1] != words[i+1][0] {
            return false
        }
    }
    return true
}

File: from-2401-to-2500/2490-circular-sentence/solution.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        if sentence[0] != sentence[len(sentence) - 1]:
            return False
        words = sentence.split(' ')
        prevWord = words[0]
        for currWord in words[1:]:
            if prevWord[len(prevWord) - 1] != currWord[0]:
                return False
            prevWord = currWord
        return True